python-avd/pyavd/schema/generate_docs/tablerowgen.py

```diff
@@ -0,0 +1,355 @@
+
+    def render_key(self) -> str:
+        """
+        Renders markdown for "key" field including mouse-over and deprecation label with color.
+        """
+        path = ".".join(self.schema._path)
+
+        if self.schema._key:
+            key = self.schema._key.replace("<", "&lt;").replace(">", "&gt;")
+        else:
+            key = f"&lt;{self.schema.type}&gt;"
+
+            # TODO: Remove legacy output
+            if LEGACY_OUTPUT:
+                path += f".{key}"
+
+        if LEGACY_OUTPUT:
+            path = path.replace("<", "&lt;").replace(">", "&gt;")
+
+        return f'[<samp>{self.get_indentation()}{key}</samp>](## "{path}"){self.get_deprecation_label()}'
+
+    def render_type(self) -> str:
+        """
+        Renders markdown for "type" field.
+        """
+        type_converters = {
+            "str": "String",
+            "int": "Integer",
+            "bool": "Boolean",
+            "dict": "Dictionary",
+            "list": "List",
+        }
+        return type_converters[self.schema.type]
+
+    def render_required(self) -> str | None:
+        """
+        Render markdown for "required" field.
+        """
+        if self.schema._is_primary_key:
+            return "Required, Unique"
+        if self.schema.required:
+            return "Required"
+
+    def render_default(self) -> str | None:
+        """
+        Should render markdown for "default" field.
+        """
+        if self.schema.default is not None:
+            if isinstance(self.schema.default, (list, dict)) and len(self.schema.default) > 1:
+                return "See (+) on YAML tab"
+
+            return f"`{self.schema.default}`"
+
+    def render_description(self) -> str | None:
+        """
+        Renders markdown for "description" field including deprecation text with color.
+        """
+        descriptions = []
+        if self.schema.description:
+            descriptions.append(self.schema.description.replace("\n", "<br>"))
+        description = " ".join(descriptions)
+
+        if deprecation := self.get_deprecation_description():
+            description += deprecation
+
+        return description or None
+
+    def render_children(self) -> Generator[TableRow]:
+        """Noop for classes without children. Override in subclasses for dict and list."""
+        yield from []
+
+    def render_restrictions(self) -> str | None:
+        """
+        Renders markdown for "restrictions" field as a multiline text compatible with a markdown table cell.
+
+        Only covers generic restrictions. Should be overridden in type specific subclasses.
+        """
+        restrictions = []
+        valid_values = []
+        if getattr(self.schema, "dynamic_valid_values", None):
+            valid_values.append(f"<value(s) of {self.schema.dynamic_valid_values}>")
+
+        if getattr(self.schema, "valid_values", None):
+            valid_values.extend(self.schema.valid_values)
```

Comment:
Could we not use our `get` function ? 
```suggestion
        valid_values.extend(get(self.schema.valid_values, []))
```

Reply:
schema is a pydantic class but I use getattr here since we don't have valid_values on dict and list types. I don't think it makes sense to introduce get() here.